tests/utils/compute_import_path_test.bzl

```diff
@@ -0,0 +1,77 @@
+# Copyright 2019 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("//proto:defs.bzl", "ProtoInfo", "proto_common", "proto_library")
+
+_TEMPLATE = """
+// Generated file. Do not edit.
+
+syntax = "proto3";
+
+package compute_import_path_test;
+
+{imports}
+""".strip()
+
+def _compute_import_path_impl(ctx):
+    imports = []
+    for dep in ctx.attr.deps:
+        file_infos = proto_common.direct_source_infos(dep[ProtoInfo])
+        imports += ['import "{}";  // From {}'.format(info.import_path, dep.label) for info in file_infos]
+
+    file = ctx.actions.declare_file("{}.proto".format(ctx.attr.name))
+    ctx.actions.write(
+        output = file,
+        content = _TEMPLATE.format(
+            imports = "\n".join(imports),
+        ),
+    )
+
+    return [
+        DefaultInfo(
+            files = depset([file]),
+        ),
+    ]
+
+_compute_import_path = rule(
+    implementation = _compute_import_path_impl,
+    attrs = {
+        "deps": attr.label_list(
+            mandatory = True,
+            providers = [ProtoInfo],
+        ),
+    },
+)
+
+def compute_import_path_test(name, deps):
+    for dep in deps:
+        target_under_test = "//tests:{}".format(dep)
```

Comment:
I'd prefer to assume deps attribute to be list of labels, and instead of constructing the label in the macro we could cut the name part of the label for gen_name. Imho that's more idiomatic this way.